sympy/solvers/inequalities.py

```diff
@@ -432,10 +433,10 @@ def valid(x):
     for x in reals:
         end = x
 
-        if ((end is S.NegativeInfinity and expr.rel_op in ['>', '>=']) or
-           (end is S.Infinity and expr.rel_op in ['<', '<='])):
-            sol_sets.append(Interval(start, S.Infinity, True, True))
-            break
+        if end in [S.NegativeInfinity, S.Infinity]:
+            if valid(S(0)):
```

Comment:
Why are you testing 0? The need to do so based on whether end is +/-oo is not transparent to me.